tools/analyzer_plugin/lib/src/diagnostic/bool_prop_name_readability.dart

```diff
@@ -28,9 +28,12 @@ class BoolPropNameReadabilityDiagnostic extends DiagnosticContributor {
         if (field.type != typeProvider.boolType) continue;
         if (propName == null) continue; // just in case
 
+        final fieldDecl = propsClass.getField(propName);
+        if (fieldDecl == null) continue;
+
         final readability = checkBoolPropReadability(propName);
         if (!readability.isReadable) {
-          collector.addError(code, result.locationFor(propsClass.getField(field.name)),
+          collector.addError(code, result.locationFor(fieldDecl),
```

Comment:
This was an unrelated runtime exception I encountered while testing this out in WSD.